test/dynamo/test_export.py

```diff
@@ -2238,17 +2238,16 @@ def foo(x):
                 return x + 2
 
         t = torch.zeros([8, 4])
-        constraints = [
```

Comment:
Can we add additional unit tests instead of changing existing ones. Though `Dim` API eventually also constructs `Constraint`s, there are still logic covered in current unit test (like dynamic_dim(x,0)>=3) that is not covered by `Dim`

Reply:
Not sure what you mean, `Dim` does have lower / upper bounds and the modified test preserves the exact behavior of the original test.

That said, there are a lot of direct unit tests for `dynamic_dim` that use `torch._dynamo.export`, which continues to use constraints. So the existing functionality of `dynamic_dim` (bounds, equalities) remains well-tested.

My purpose in modifying tests is two-fold: (a) make sure `Dim` has good coverage (b) avoid warnings in tests, which is a strange thing to have. 

Finally, once we decide to remove `constraints` I would like to remove `dynamic_dim` entirely, since it is way more direct to go from `Dim` to symbols instead of going through `dynamic_dim`. So this work needed to be done anyway.